[Ang Jason] iP

kdeasymoneysniper opened the pull request:
# Duke Bot

Here's what this program *can* do:
* add tasks
* manage tasks
* and everything else

Steps to use this bot:
1. leave a good review on this pull request
2. repeat step 1


**Todo**
- [ ] weekly iP
- [x] oh wait why am i using this when i have Duke Bot? 😉 


You are on [GitHub](https://github.com)

Here's some sample Java code with the `main` method.
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

> "Had a dream I was king, I woke up still king."

Commit:
Add Gradle support

kdeasymoneysniper committed:
changes

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

rushilramesh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+        try {
+            FileInputStream fis = new FileInputStream(filePath);
+            Scanner sc = new Scanner(fis);
+            while (sc.hasNextLine()) {
```
Perhaps you could use a switch statement here?

rushilramesh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,138 @@
+            System.out.println(myList.get(myList.size() - 1));
+            System.out.println("Now you have " + myList.size() + " tasks.");
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Invalid format. Please follow this format. <type of task> <description> /<date if necessary>");
```
Perhaps you could throw a DukeException (or create an exceptions class) and catch it in Duke.java's run function instead of printing a different output for every case?

rushilramesh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,138 @@
+
+    private void addTask(String input) {
+        try {
+            if (input.startsWith("todo")) {
```
Perhaps you could use an enum to represent the different tasks and a switch statement here?

rushilramesh commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,42 @@
+        String output = "[T][ ] task1\n";
+        assertEquals(tasks.output(), output);
+    }
+}
```
I like your use of meaningful test cases.

rushilramesh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,47 @@
+            } else {
+                tasks.action(input);
+                storage.saveTasks(tasks.output());
+            }
```
I like the logical flow of the Duke program

rushilramesh reviewed (commented):
LGTM. Good work and atb for the future levels!

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
Merge branch 'add-gradle-support'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
Add JAR File

A JAR File makes it easier and more efficient to deploy and
distribute an entire program, which is why it is added in this version.

Used Gradle to create a JAR File that includes the
classes and resources, along with the JavaFX libraries

Done using Gradle as it is an easy way to do so, and it is the
instruction given to us.

kdeasymoneysniper committed:
Add Assertions to code

Assertions help to ensure the program runs as intended.

A few assertions are added to ensure that code logic does not fail.

Asserts are helpful in making sure there are no loopholes.

kdeasymoneysniper committed:
Improve Code Quality

The exception handling was sloppy before, with tons of catch statements

DukeException class is created to account for exceptions, although
there are still multiple catch statements around, but would hopefully
be solved in future versions

Having a central place to catch exceptions make the code easier to read.

kdeasymoneysniper committed:
Merge pull request #1 from kdeasymoneysniper/branch-A-Assertions

Add Assertions to code

kdeasymoneysniper committed:
Merge branch 'master' into branch-A-CodeQuality

kdeasymoneysniper committed:
Merge pull request #2 from kdeasymoneysniper/branch-A-CodeQuality

Improve Code Quality

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes'

kdeasymoneysniper committed:
'changes

kdeasymoneysniper committed:
'changes'
